From 0293cc6eba120a342c563fc23b9b53f8affad8b7 Mon Sep 17 00:00:00 2001
From: Georgia Garcia <georgia.garcia@canonical.com>
Date: Tue, 18 Oct 2022 10:37:34 -0300
Subject: [PATCH] add 2.13 abi and pin it in parser.conf

Policies that were created before adding new mediation will not
be affected if we pin the abi on parser.conf.

Signed-off-by: Georgia Garcia <georgia.garcia@canonical.com>
---
 parser/parser.conf           |  3 ++
 profiles/apparmor.d/abi/2.13 | 80 ++++++++++++++++++++++++++++++++++++
 utils/apparmor/aa.py         |  2 +-
 3 files changed, 84 insertions(+), 1 deletion(-)
 create mode 100644 profiles/apparmor.d/abi/2.13

diff --git a/parser/parser.conf b/parser/parser.conf
index 641cf1db8..93bb9d458 100644
--- a/parser/parser.conf
+++ b/parser/parser.conf
@@ -59,3 +59,6 @@
 ## Adjust compression
 #Optimize=compress-small
 #Optimize=compress-fast
+
+## Pin older policies so they are not affected by new mediation
+policy-features=/etc/apparmor.d/abi/2.13
diff --git a/profiles/apparmor.d/abi/2.13 b/profiles/apparmor.d/abi/2.13
new file mode 100644
index 000000000..3463f372a
--- /dev/null
+++ b/profiles/apparmor.d/abi/2.13
@@ -0,0 +1,80 @@
+capability {0xffffff
+}
+caps {mask {chown dac_override dac_read_search fowner fsetid kill setgid setuid setpcap linux_immutable net_bind_service net_broadcast net_admin net_raw ipc_lock ipc_owner sys_module sys_rawio sys_chroot sys_ptrace sys_pacct sys_admin sys_boot sys_nice sys_resource sys_time sys_tty_config mknod lease audit_write audit_control setfcap mac_override mac_admin syslog wake_alarm block_suspend audit_read perfmon bpf checkpoint_restore
+}
+}
+dbus {mask {acquire send receive
+}
+}
+domain {attach_conditions {xattr {yes
+}
+}
+change_hat {yes
+}
+change_hatv {yes
+}
+change_onexec {yes
+}
+change_profile {yes
+}
+computed_longest_left {yes
+}
+fix_binfmt_elf_mmap {yes
+}
+post_nnp_subset {yes
+}
+stack {yes
+}
+version {1.2
+}
+}
+file {mask {create read write exec append mmap_exec link lock
+}
+}
+mount {mask {mount umount pivot_root
+}
+}
+namespaces {pivot_root {no
+}
+profile {yes
+}
+}
+network {af_mask {unspec unix inet ax25 ipx appletalk netrom bridge atmpvc x25 inet6 rose netbeui security key netlink packet ash econet atmsvc rds sna irda pppox wanpipe llc ib mpls can tipc bluetooth iucv rxrpc isdn phonet ieee802154 caif alg nfc vsock kcm qipcrtr smc xdp mctp
+}
+af_unix {yes
+}
+}
+network_v8 {af_mask {unspec unix inet ax25 ipx appletalk netrom bridge atmpvc x25 inet6 rose netbeui security key netlink packet ash econet atmsvc rds sna irda pppox wanpipe llc ib mpls can tipc bluetooth iucv rxrpc isdn phonet ieee802154 caif alg nfc vsock kcm qipcrtr smc xdp mctp
+}
+}
+policy {outofband {0x000001
+}
+set_load {yes
+}
+versions {v5 {yes
+}
+v6 {yes
+}
+v7 {yes
+}
+v8 {yes
+}
+}
+}
+ptrace {mask {read trace
+}
+}
+query {label {data {yes
+}
+multi_transaction {yes
+}
+perms {allow deny audit quiet
+}
+}
+}
+rlimit {mask {cpu fsize data stack core rss nproc nofile memlock as locks sigpending msgqueue nice rtprio rttime
+}
+}
+signal {mask {hup int quit ill trap abrt bus fpe kill usr1 segv usr2 pipe alrm term stkflt chld cont stop stp ttin ttou urg xcpu xfsz vtalrm prof winch io pwr sys emt lost
+}
+}
diff --git a/utils/apparmor/aa.py b/utils/apparmor/aa.py
index f9cd871e8..d4a30005f 100644
--- a/utils/apparmor/aa.py
+++ b/utils/apparmor/aa.py
@@ -2048,7 +2048,7 @@ def is_skippable_file(path):
     return False
 
 def is_skippable_dir(path):
-    if re.search('^(.*/)?(disable|cache|cache\.d|force-complain|lxc|\.git)/?$', path):
+    if re.search('^(.*/)?(disable|cache|cache\.d|force-complain|lxc|abi|\.git)/?$', path):
         return True
     return False
 
-- 
2.34.1

